### ISSUE TITLE
Temperature Only Displays in C

### ISSUE THREAD

**Issue by Casey-Sheridan** [petrleocompel/hikaxpro_hacs#139]
Prior to taking update 2.3.0, the sensor temperatures were able to display in whatever unit was desired (F for those of us in the US). I can say for sure which version this changed in, but now the temperature is only available in C.![image](https://github.com/user-attachments/assets/9f630225-da8d-459b-abb2-e8cd84051ccf)![image](https://github.com/user-attachments/assets/15911584-0ef6-424f-8897-306606436e1a)

**Comment by petrleocompel** [comment]
Device sends info in metric units. If you want different units you can use a helpers as workaround.

**Comment by Casey-Sheridan** [comment]
> Device sends info in metric units. If you want different units you can use a helpers as workaround. 

This is true. I was just curious at the sudden change and figured that there must have been something in the code that was adjusted.

**Comment by petrleocompel** [comment]
Any change in your user profile ? There is setting for "temperature units".

For temperature sensors we have `self._attr_native_unit_of_measurement = UnitOfTemperature.CELSIUS` -> `NATIVE`
So HA should convert it based on your user profile preference.

**Comment by petrleocompel** [comment]
Can you check on that ? @Casey-Sheridan

**Comment by Casey-Sheridan** [comment]
I checked around and couldn't find anything in the profile referencing Celsius as the primary unit. To my recollection, I don't believe any profile changes were made at the time, unless it had to do with an upgrade to HA.

**Comment by petrleocompel** [comment]
Sorry my mistake it is system wide setting in "system" 
<img width="1209" alt="image" src="https://github.com/user-attachments/assets/d83e897d-70d1-4772-be32-8735570d4cd3" />

**Comment by petrleocompel** [comment]
Is your setup correct ? @Casey-Sheridan

**Comment by petrleocompel** [comment]
Found root cause of this - Temp sensors need to have "measurement" unit in code. Can be fixed in next release

**Comment by petrleocompel** [comment]
Temperature unit conversion is fixed by using `_attr_device_class` correctly.

Fixed in **v3.2.0**. Please upgrade via HACS (or update the custom component) and reload the integration. Reopen if it still reproduces on 3.2.0.
